src/client/components/pages/Courses/MeetingModal.tsx

```diff
@@ -279,8 +268,19 @@ const MeetingModal: FunctionComponent<MeetingModalProps> = function ({
     setCurrentEditMeeting,
     setShowRoomsData,
     isChanged,
-    initialized,
-    setInitialized,
+  ]);
+
+  /**
+   * Only set the meetings in the modal if the modal is visible and the
+   * meetings haven't already yet been set.
+   */
+  useEffect(() => {
+    setAllMeetings(instanceMeetings);
```

Comment:
This still needs to be wrapped in `if (isVisible) { ... }`, otherwise it'll fire off a bunch of state updates on closed modals when the page loads.

Reply:
Oops! Thanks for catching that. This was fixed in 4ad8b38.